consensus/istanbul/backend/engine.go

```diff
@@ -646,6 +646,10 @@ func (sb *Backend) StartValidating(hasBadBlock func(common.Hash) bool,
 		headBlock := sb.GetCurrentHeadBlock()
 		valset := sb.getValidators(headBlock.Number().Uint64(), headBlock.Hash())
 		sb.RefreshValPeers(valset)
+		// For a proxied validator, this is called in `sb.addProxy`
+		if err := sb.updateAnnounceVersion(newAnnounceVersion()); err != nil {
```

Comment:
Does this need to be done here?  Seems like the proxy is sent the enodeCertificate in `addPeer`.

Reply:
This call is for a non proxied validator, but the benefit of doing this here for a non-proxied validator is that `sb.enodeCertificateMsg` will be created without having to wait for the first announce msg to be generated/gossiped (iirc that takes about a minute). This allows the validator to prove itself during the handshake if needed during that time (& at startup I see this being pretty important)